txpool/txpool.go

```diff
@@ -881,6 +898,49 @@ func (p *TxPool) resetAccounts(stateNonces map[types.Address]uint64) {
 	}
 }
 
+// updateUnadoptedCounts update the accounts' unadopted counts,
+// the number of the consecutive blocks that doesn't have the account's transactions
+func (p *TxPool) updateUnadoptedCounts(latestActiveAccounts map[types.Address]uint64) {
+	p.accounts.Range(
+		func(key, value interface{}) bool {
+			address, _ := key.(types.Address)
+			account, _ := value.(*account)
+
+			if _, ok := latestActiveAccounts[address]; ok {
+				account.unadopted = 0
+			} else {
+				account.unadopted++
+			}
+
+			if account.unadopted < maxAccountUnadopted {
+				return true
+			}
+
+			account.promoted.lock(true)
+			defer account.promoted.unlock()
+
+			firstTx := account.promoted.pop()
+			if firstTx != nil {
+				p.index.remove(firstTx)
+				p.gauge.decrease(slotsRequired(firstTx))
+				account.unadopted = 0
+			}
+
+			account.enqueued.lock(true)
+			defer account.enqueued.unlock()
+
+			firstTx = account.enqueued.pop()
+			if firstTx != nil {
+				p.index.remove(firstTx)
+				p.gauge.decrease(slotsRequired(firstTx))
+				account.unadopted = 0
+			}
```

Comment:
This is the critical part the txpool needs to do correctly: since we're trying to mimic what validator nodes do when performing the `Demote->Drop` this piece of code should be replaced with a call to `Drop()` 

(Drop does all of this logic already and updates the metrics as well)

The core idea is symmetrical - `demotions` field is for validators `skips` (unadopted) for non-validators, but the behavior needs to be identical.